### PR TITLE
Istio reconciler reset proxy more resielient, verify Istio install

### PR DESCRIPTION
**Pull request by videlov**
- Add Istio install verification step to check if istio-system pods are running with the new target version after installation.
- Add additional check before resetting proxies, checks if current Istio version is the target version for proxy reset.

### CODE REVIEW

**Review by jeremyharisch — DISMISSED** [review]

File: pkg/reconciler/instances/istio/action.go [hunk 1]
```diff
@@ -274,13 +276,46 @@ func canUpdate(istioStatus actions.IstioStatus) (bool, error) {
...
+	for dpVersion := range istioStatus.DataPlaneVersions {
+		if isDataplaneCompatible, err := isComponentCompatible(dpVersion, istioStatus.TargetVersion, "Data plane"); !isDataplaneCompatible {
+			return false, err
+		}
+	}
+
+	return true, nil
+}
+
+func canResetProxies(istioStatus actions.IstioStatus) (bool, error) {
```
Comment (cnvergence) [inline pkg/reconciler/instances/istio/action.go:288]: do we need to return bool here?

File: pkg/reconciler/instances/istio/action.go [hunk 2]
```diff
@@ -139,15 +141,15 @@ func (a *ProxyResetPostAction) Run(context *service.ActionContext) error {
 		return err
 	}
 
-	canUpdateResult, err := canUpdate(istioStatus)
+	_, err = canResetProxies(istioStatus)
```
Comment (cnvergence) [inline pkg/reconciler/instances/istio/action.go:144]: can we somehow use this bool?

**Comment by cnvergence** [comment]
LGTM after fixes, just left small comments

File: pkg/reconciler/instances/istio/action.go (hunk 1 above)
Comment (videlov) [reply pkg/reconciler/instances/istio/action.go]: actually no need for the bool, indeed.

File: pkg/reconciler/instances/istio/action.go (hunk 2 above)
Comment (videlov) [reply pkg/reconciler/instances/istio/action.go]: removed it.

**Review by cnvergence — APPROVED** [review]

**Review by jeremyharisch — APPROVED** [review]

**Pull request merged**